packages/flutter/lib/src/widgets/text_editing_action.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'actions.dart';
+import 'editable_text.dart';
+import 'focus_manager.dart';
+import 'framework.dart';
+
+/// Similar to [CallbackAction]'s [OnInvokeCallback], but includes
+/// [EditableTextState] as a parameter.
+typedef OnInvokeTextEditingCallback<T extends Intent> = Object? Function(T intent, EditableTextState editableTextState);
+
+/// An [Action] related to editing text.
+///
+/// If an [EditableText] is currently focused, then it will be passed to the
+/// given [onInvoke] callback. If not, then [isEnabled] will be false and
+/// [onInvoke] will not be called.
+///
+/// See also:
+///
+///  * [CallbackAction], which is a similar Action type but unrelated to text
+///    editing.
+class TextEditingAction<T extends Intent> extends ContextAction<T> {
```

Comment:
Wondering: Should this just be an abstract class that you need to extend to implement the abstract onInvoke method?

(not feeling super strongly about this, though)

Reply:
I actually really liked this idea so I went for it in e24610a.  I didn't like committing to a new signature of onInvoke, and this lets me get rid of it altogether.  I think it will be more flexible this way.